Tools/AP_Periph/relay.cpp

```diff
@@ -0,0 +1,31 @@
+#include "AP_Periph.h"
+
+#if AP_RELAY_ENABLED
+#include <dronecan_msgs.h>
+
+void AP_Periph_FW::handle_hardpoint_command(CanardInstance* canard_instance, CanardRxTransfer* transfer)
+{
+    uavcan_equipment_hardpoint_Command cmd {};
+    if (uavcan_equipment_hardpoint_Command_decode(transfer, &cmd)) {
+        // Failed to decode
+        return;
+    }
+
+    // Translate hardpoint ID to relay function
+    AP_Relay_Params::FUNCTION fun;
+    switch (cmd.hardpoint_id) {
+        case 0 ... 15:
+            // 0 to 15 are continuous
+            fun = AP_Relay_Params::FUNCTION(cmd.hardpoint_id + (uint8_t)AP_Relay_Params::FUNCTION::DroneCAN_HARDPOINT_0);
+            break;
+
+        default:
+            // ID not supported
+            return;
+    }
+
+    // Set relay
+    relay.set(fun, cmd.command);
```

Comment:
This should be stricter in what it passes through IMO.

i.e. don't throw whatever at the relay-set function in case we *do* want to add stuff later and can't afford to stuff up existing periphs

Reply:
Now checks the command is 0 or 1.